File: LeetCode/Medium/0443-string-compression/0443-string-compression.cs
public class Solution {
    public int Compress(char[] chars) {
        var r = 0;
        var w = 0;

        while (r < chars.Length)
        {
            var c = r;

            while (r < chars.Length && chars[r] == chars[c]) r++;
            
            chars[w++] = chars[c];
            var count = r - c;
            if (count > 1)
            {
                count.TryFormat(chars.AsSpan(w), out int charsWritten);
                w += charsWritten;
            }
        }

        return w;
    }
}

File: LeetCode/Medium/0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:          
        read_idx = 0
        write_idx = 0
        while read_idx < len(chars):
            anchor = read_idx
            while read_idx < len(chars) and chars[read_idx] == chars[anchor]:
                read_idx += 1
            
            chars[write_idx] = chars[anchor]
            write_idx += 1
            group_length = read_idx - anchor
            if group_length > 1:
                count = str(group_length)

                for c in count:
                    chars[write_idx] = c
                    write_idx += 1

        return write_idx